tests/integrationtest/run-tests.sh

```diff
@@ -58,19 +58,54 @@ function help_message()
                     This option will be ignored if \"-r <test-name>\" is provided.
                     Run all tests if this option is not provided.
 
-    -p <portgenerator-path>: Use port generator in <portgenerator-path> for generating port numbers.
-
 "
 }
 
-function build_portgenerator()
-{
-    portgenerator="./portgenerator"
-    echo "building portgenerator binary: $portgenerator"
-    rm -rf $portgenerator
-    GO111MODULE=on go build -o $portgenerator github.com/pingcap/tidb/cmd/portgenerator
+# Function to find an available port starting from a given port
+function find_available_port() {
+    local start_port=$1
+    local port=$start_port
+
+    while :; do
+        if command -v ss &> /dev/null; then
```

Comment:
Does our docker image has `ss` command?

Reply:
Maybe we could read `/proc/net/tcp` file to get it.

Reply:
I'll use `lsof` instead.

Reply:
Not sure if our docker has it